### PR TITLE
core: pivot all-day dates assuming the timezone of the resource

### PR DESCRIPTION
**Pull request by mworrell**
### Description

This makes some changes to the way "all day" date ranges are pivoted.

The problem was:

 * Date is in CET and all day
 * Pivot is stored in UTC (without date conversion)
 * Search for date in CET doesn't return the resource, as it is shifted one (or two for CEST) hours.

Solution:

 * Pivot the "all day" date_start and date_end shifted from the resource timezone to UTC.

This pull request changes the behavior for pivoting (and searching) the "all day" date ranges.
Now the date for all-day date ranges is stored as if that date is in the timezone stored with the resource.
This timezone is in the `tz` property and will default to the timezone of the editor creating the resource.

If the `tz` property is not set, then the default Zotonic timezone is used.

Note that the way the date itself is stored in `date_start` and `date_end` is unaffected.

Also adds an option to the `date` and `date_range` filters to pass an integer as the timezone.
This is assumed to be a resource id, and the date is then displayed in the timezone of the resource's `tz` property.

**WARNING**

This introduces an **incompatibilty** where previously passing `1` as the timezone will use `UTC`. From now on pass `true` or `"UTC"` to get a time in UTC.   Passing `1` will use the timezone in the `tz` property of the resource with id 1.

### Checklist

- [x] documentation updated
- [ ] tests added
- [ ] no BC breaks

### CODE REVIEW

**Comment by robvandenbogaard** [comment]
I guess the timezone of "administrator" (resource id 1) would not be UTC by default? (probably Zotonic zone instead) Because if so, the effect of the incompatibility of passing 1 would be limited.

**Review by mmzeeman — APPROVED** [review]

File: apps/zotonic_core/src/support/z_pivot_rsc_job.erl [hunk 1]
```diff
@@ -71,15 +71,36 @@ pivot_resource_update(Id, UpdateProps, RawProps, Context) ->
...
+    Date;
+tz_all_day(true, ?ST_JUTTEMIS = Date, _Context) ->
+    Date;
+tz_all_day(true, Date, Tz) when is_tuple(Date) ->
+    % All-day dates are stored without timezone conversion.
+    % For indexing we store them as-if they were entered in the
+    % timezone of the resource.
+    z_datetime:to_utc(Date, Tz);
+tz_all_day(_IsAllDay, Date, _Context) ->
+    Date.
```
Comment (robvandenbogaard) [inline apps/zotonic_core/src/support/z_pivot_rsc_job.erl:103]: The current name for this function suggests it returns a tz; I think I'd rather call it something like `tz_shift_all_day`.

Comment (mworrell) [reply apps/zotonic_core/src/support/z_pivot_rsc_job.erl:103]: Agree - will change

**Comment by robvandenbogaard** [comment]
> @robvandenbogaard The `1` and the timezone might be problematic indeed. We could also log a warning and then just continue with whatever `1` has as a tz.
> 
> This is possible as this is a new feature and I don't expect it to be used immediately. Especially not for id 1.

Good to note there is a workaround in case `1` causes trouble - setting admin tz to utc.

**Review by robvandenbogaard — APPROVED** [review]

**Pull request merged**